src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -0,0 +1,181 @@
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RAY_GCS_ACTOR_MANAGER_H
+#define RAY_GCS_ACTOR_MANAGER_H
+
+#include <ray/common/id.h>
+#include <ray/common/task/task_execution_spec.h>
+#include <ray/common/task/task_spec.h>
+#include <ray/gcs/redis_gcs_client.h>
+#include <ray/protobuf/gcs_service.pb.h>
+#include <queue>
+#include <utility>
+
+#include "gcs_actor_scheduler.h"
+
+namespace ray {
+namespace gcs {
+
+/// GcsActor just wraps `ActorTableData` and provides some convenient interfaces to access
+/// the fields inside `ActorTableData`.
+class GcsActor {
+ public:
+  /// Create a GcsActor
+  ///
+  /// \param actor_table_data Data of the actor (see gcs.proto).
+  /// \param redis_gcs_client The client to communicate with the redis/storage, it will be
+  /// replaced with StorageClient in the future.
+  explicit GcsActor(rpc::ActorTableData &&actor_table_data,
+                    std::shared_ptr<RedisGcsClient> redis_gcs_client)
+      : actor_table_data_(std::move(actor_table_data)),
+        redis_gcs_client_(std::move(redis_gcs_client)) {}
+
+  /// Get the node id on which this actor is created.
+  ClientID GetNodeID() const;
+  /// Get the id of the worker on which this actor is created.
+  WorkerID GetWorkerID() const;
+
+  /// Update the `Address` of this actor (see gcs.proto).
+  void UpdateAddress(const rpc::Address &address);
+  /// Get the `Address` of this actor.
+  const rpc::Address &GetAddress() const;
+
+  /// Update the state of this actor.
+  void UpdateState(rpc::ActorTableData::ActorState state);
+  /// Get the state of this gcs actor.
+  rpc::ActorTableData::ActorState GetState() const;
+
+  /// Get the id of this actor.
+  ActorID GetActorID() const;
+  /// Get the task specification of this actor.
+  TaskSpecification GetCreationTaskSpecification() const;
+
+  /// Get the immutable ActorTableData of this actor.
+  const rpc::ActorTableData &GetActorTableData() const;
+  /// Get the mutable ActorTableData of this actor.
+  rpc::ActorTableData *GetMutableActorTableData();
+
+  /// Flush the current actor meta info to storage asynchronously.
+  ///
+  /// \param callback Invoked when flushed successfully.
+  void AsyncFlush(std::function<void()> &&callback = nullptr);
+
+ private:
+  /// The actor meta data which contains the task specification as well as the state of
+  /// the gcs actor and so on (see gcs.proto).
+  rpc::ActorTableData actor_table_data_;
+  /// The client to communicate with the redis/storage, it will be replaced with
+  /// StorageClient in the future.
+  std::shared_ptr<RedisGcsClient> redis_gcs_client_;
```

Comment:
> It doesn't really make sense to store the client in the `GcsActor` since the description says that this is just a wrapper class, but the `RedisGcsClient` adds a significant dependency. I notice it's only used for the `AsyncFlush` method; I would just make that method an anonymous helper function instead.

It's a good suggestion, thanks!

Reply:
> It doesn't really make sense to store the client in the `GcsActor` since the description says that this is just a wrapper class, but the `RedisGcsClient` adds a significant dependency. I notice it's only used for the `AsyncFlush` method; I would just make that method an anonymous helper function instead.

In fact, there is no such method at the beginning. The flush operations of the actor are managed by the `GcsActorManager`. Because the `GcsActorsSheduler` also needs to flush actor when its address changes, so these two classes are interdependent. To resolve the interdependence, let the actor rely on the `RedisGcsClient` directly.